static/js/protein/chart.ts

```diff
@@ -73,6 +74,13 @@ export interface VarGeneDataPoint {
   upper: number;
 }
 
+type Selectable = Element | EnterElement | Document | Window | SVGLineElement;
```

Comment:
Is this necessary? I see it's used when you pass a selection to handleMouseEvents, but aren't all the selections going to be of type `Element`?

Reply:
Oh yea I couldn't figure this one out - if I remove any of these I get an opaque error of the form

```
[tsl] ERROR in /home/rogershijin/project/datacommons-website/static/js/protein/chart.ts(838,7)
      TS2345: Argument of type '(selection: Selection<Element, Datum, SVGGElement, unknown>, idFunc: (index: number) => string, toolTipFunc: (datum: Datum) => string, brightenPercentage?: string) => void' is not assignable to parameter of type '(selection: Selection<SVGLineElement | BaseType, InteractionLink, SVGGElement, unknown>, ...args: any[]) => void'.
  Types of parameters 'selection' and 'selection' are incompatible.
    Type 'Selection<SVGLineElement | BaseType, InteractionLink, SVGGElement, unknown>' is not assignable to type 'Selection<Element, Datum, SVGGElement, unknown>'.
      Type 'SVGLineElement | BaseType' is not assignable to type 'Element'.
        Type 'Window' is missing the following properties from type 'Element': attributes, classList, className, clientHeight, and 154 more.
```

The current hack uses the fact that a [union of types only guarantees the intersection of their properties](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types) to deal with this, but perhaps there's a much more straightforward way to do this.  Any tips would be greatly appreciated.

Reply:
Wow this works!!! Brilliant!

I also experimented with the below things which failed - I would've never thought to modify the latter 3 arguments without your example.

```
d3.Selection<SVGElement, Datum, SVGGElement, unknown>
d3.Selection<SVGElement | SVGLineElement, Datum, SVGGElement, unknown>
d3.Selection<Element | SVGElement | SVGLineElement, Datum, SVGGElement, unknown>
```